### DISCUSSION TITLE
How to create a server side code outside of /pages/api or getServerSideProps?

### DISCUSSION THREAD

**Discussion by v3rron** [vercel/next.js#32783] [App Router]
Hi there,

I looked everywhere in the documentation and either I'm blind or can someone explain how do I write a code that is only accessible to the server. 

We're working on a project with external API that I need to connect to and access httpOnly cookies for authorization. I tried putting the the /lib folder, but apparently it's a client side folder that has no access to the server or server-side cookies and exposes the environment variables of security credentials to the client.

So far in all the examples everyone uses /pages/api for consuming external APIs, but that approach also exposes all the API routes to the client, which I don't want to do.

I'm basically looking for an equivalent of a `/server` folder in Meteor, where you can just put any critical server code without conventions.

Thanks in advance.

**Comment by tahonaPL** [comment]
I have similar problem: https://github.com/vercel/next.js/discussions/32874
